### PR TITLE
Adding support to cpuid.py for AMD Hygon Dhyana (new)

### PR DESCRIPTION
**Pull request by mreed8855**
## Description
This provides support for the  AMD Hygon Dhyana CPU in the cpuid.py script.

## Resolved issues
https://github.com/canonical/checkbox/issues/529
SERVCERT-1147

### CODE REVIEW

File: providers/base/bin/cpuid.py [hunk 1]
```diff
@@ -112,6 +112,7 @@
         "AMD ROME":         ['0x830f10'],
         "AMD Ryzen":        ['0x810f81'],
         "AMD Bergamo":      ['0xaa0f01'],
+        "AMD Hygon Dhyana"  ['0x900f22'],
```
Comment (bladernr) [inline providers/base/bin/cpuid.py:115]: Don't call it an AMD, just "Hygon Dhyana Plus"
(I just looked at the info and apparently the 32 core is Dhyana Plus and there's a smaller 8-core thats just Dhyana).

**Review by bladernr — CHANGES_REQUESTED** [review]
just a minor change needed and it's good to go.

**Review by bladernr — APPROVED** [review]

**Review by bladernr — CHANGES_REQUESTED** [review]
You're missing a colon, causing the flake8 test to fail.

So since you need to fix that can you also move the entry so it's in alphabetic order with the rest of the list? I wasn't going to worry about that but since you need to add that colon anyway, may as well deal with that too.

**Comment by mreed8855** [comment]
My apologies for the typo.  I have added the colon and I had to abbreviate the name to keep the formatting consistent.

**Comment by kissiel** [comment]
TBH the formatting is already wonky, I'd prefer reformatting the whole dict to a proper python code instead, abbreviating "plus" to "pls" to keep that formatting.
In other words:
```
CPUIDS = {
-        "Amber Lake":       ['0x806e9'],
-        "AMD EPYC":         ['0x800f12'],
-        "AMD Genoa":        ['0xa10f11'],
-        "AMD Lisbon":       ['0x100f81'],
-        "AMD Magny-Cours":  ['0x100f91'],
-        "AMD Milan":        ['0xa00f11'],
-        "AMD Milan-X":      ['0xa00f12'],
-        "AMD ROME":         ['0x830f10'],
-        "AMD Ryzen":        ['0x810f81'],
-        "AMD Bergamo":      ['0xaa0f01'],
-        "Hygon Dhyana Pls": ['0x900f22'],
-        "Broadwell":        ['0x4067', '0x306d4', '0x5066', '0x406f'],
-        "Canon Lake":       ['0x6066'],
-        "Cascade Lake":     ['0x50655', '0x50656', '0x50657'],
-        "Coffee Lake":      [
-            '0x806ea', '0x906ea', '0x906eb', '0x906ec', '0x906ed'],
-        "Comet Lake":       ['0x806ec', '0xa065'],
-        "Cooper Lake":      ['0x5065a', '0x5065b'],
-        "Haswell":          ['0x306c', '0x4065', '0x4066', '0x306f'],
-        "Ice Lake":         ['0x606e6', '0x606a6', '0x706e6', '0x606c1'],
-        "Ivy Bridge":       ['0x306a', '0x306e'],
-        "Kaby Lake":        ['0x806e9', '0x906e9'],
-        "Knights Landing":  ['0x5067'],
-        "Knights Mill":     ['0x8065'],
-        "Nehalem":          ['0x106a', '0x106e5', '0x206e'],
-        "Pineview":         ['0x106ca'],
-        "Penryn":           ['0x1067a'],
-        "Rocket Lake":      ['0xa0671'],
-        "Sandy Bridge":     ['0x206a', '0x206d6', '0x206d7'],
-        "Sapphire Rapids":  ['0x806f3', '0x806f6', '0x806f7', '0x806f8'],
-        "Skylake":          ['0x406e3', '0x506e3', '0x50654', '0x50652'],
-        "Tiger Lake":       ['0x806c1'],
-        "Aderlake":         ['0x906a4', '0x906A3', '0x90675', '0x90672'],
-        "Raptorlake":       ['0xB0671'],
-        "Westmere":         ['0x2065', '0x206c', '0x206f'],
-        "Whisky Lake":      ['0x806eb', '0x806ec'],
-        }
+    "Amber Lake": ["0x806e9"],
+    "AMD EPYC": ["0x800f12"],
+    "AMD Genoa": ["0xa10f11"],
+    "AMD Lisbon": ["0x100f81"],
+    "AMD Magny-Cours": ["0x100f91"],
+    "AMD Milan": ["0xa00f11"],
+    "AMD Milan-X": ["0xa00f12"],
+    "AMD ROME": ["0x830f10"],
+    "AMD Ryzen": ["0x810f81"],
+    "AMD Bergamo": ["0xaa0f01"],
+    "Hygon Dhyana Plus": ["0x900f22"],
+    "Broadwell": ["0x4067", "0x306d4", "0x5066", "0x406f"],
+    "Canon Lake": ["0x6066"],
+    "Cascade Lake": ["0x50655", "0x50656", "0x50657"],
+    "Coffee Lake": ["0x806ea", "0x906ea", "0x906eb", "0x906ec", "0x906ed"],
+    "Comet Lake": ["0x806ec", "0xa065"],
+    "Cooper Lake": ["0x5065a", "0x5065b"],
+    "Haswell": ["0x306c", "0x4065", "0x4066", "0x306f"],
+    "Ice Lake": ["0x606e6", "0x606a6", "0x706e6", "0x606c1"],
+    "Ivy Bridge": ["0x306a", "0x306e"],
+    "Kaby Lake": ["0x806e9", "0x906e9"],
+    "Knights Landing": ["0x5067"],
+    "Knights Mill": ["0x8065"],
+    "Nehalem": ["0x106a", "0x106e5", "0x206e"],
+    "Pineview": ["0x106ca"],
+    "Penryn": ["0x1067a"],
+    "Rocket Lake": ["0xa0671"],
+    "Sandy Bridge": ["0x206a", "0x206d6", "0x206d7"],
+    "Sapphire Rapids": ["0x806f3", "0x806f6", "0x806f7", "0x806f8"],
+    "Skylake": ["0x406e3", "0x506e3", "0x50654", "0x50652"],
+    "Tiger Lake": ["0x806c1"],
+    "Aderlake": ["0x906a4", "0x906A3", "0x90675", "0x90672"],
+    "Raptorlake": ["0xB0671"],
+    "Westmere": ["0x2065", "0x206c", "0x206f"],
+    "Whisky Lake": ["0x806eb", "0x806ec"],
+}
```

**Review by bladernr — DISMISSED** [review]

**Comment by kissiel** [comment]
@bladernr I got no problems with the file, just seeing that Michael had to do cut a character made me write that 😄

**Review by kissiel — DISMISSED** [review]
All looks good!
+1

**Comment by bladernr** [comment]
@pieqq what is "black" and why was it run against this? It seems to have munged all the formatting and triggered several of codecov complaints.

For example:
`    0x89, 0x5f, 0x04,        # mov    %ebx,0x4(%rdi)`

is now:
```
    0x89,
    0x5F,
    0x04,  # mov    %ebx,0x4(%rdi)
```
which may be functional but lost a lot of the explanation in comment.

**Pull request closed without merging**